core/src/main/java/org/mule/runtime/core/internal/processor/strategy/TransactionAwareProactorStreamEmitterProcessingStrategyFactory.java

```diff
@@ -0,0 +1,138 @@
+import org.mule.runtime.core.internal.processor.strategy.ProactorStreamEmitterProcessingStrategyFactory.ProactorStreamEmitterProcessingStrategy;
+import org.mule.runtime.core.internal.util.rx.ConditionalExecutorServiceDecorator;
+
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+/**
+ * Creates default processing strategy with same behavior as {@link ProactorStreamEmitterProcessingStrategyFactory} apart from the
+ * fact it will process synchronously without error when a transaction is active.
+ *
+ * @since 4.0
+ */
+public class TransactionAwareProactorStreamEmitterProcessingStrategyFactory extends ReactorStreamProcessingStrategyFactory
+    implements TransactionAwareProcessingStrategyFactory {
+
+  @Override
+  public ProcessingStrategy create(MuleContext muleContext, String schedulersNamePrefix) {
+    return new TransactionAwareProactorStreamEmitterProcessingStrategy(getRingBufferSchedulerSupplier(muleContext,
+                                                                                                      schedulersNamePrefix),
+                                                                       getBufferSize(),
+                                                                       getSubscriberCount(),
+                                                                       getWaitStrategy(),
+                                                                       () -> muleContext.getSchedulerService()
+                                                                           .cpuLightScheduler(muleContext.getSchedulerBaseConfig()
+                                                                               .withName(schedulersNamePrefix + "."
+                                                                                   + CPU_LITE.name())),
+                                                                       () -> muleContext.getSchedulerService()
+                                                                           .ioScheduler(muleContext.getSchedulerBaseConfig()
+                                                                               .withName(schedulersNamePrefix + "."
+                                                                                   + BLOCKING.name())),
+                                                                       () -> muleContext.getSchedulerService()
+                                                                           .cpuIntensiveScheduler(muleContext
+                                                                               .getSchedulerBaseConfig()
+                                                                               .withName(schedulersNamePrefix + "."
+                                                                                   + CPU_INTENSIVE.name())),
+                                                                       () -> muleContext.getSchedulerService()
```

Comment:
this code to get the retry support scheduler is outdated. Copy the code from the previous PS implementation.